src/main/java/com/redhat/developers/msa/aloha/AlohaVerticle.java

```diff
@@ -40,47 +44,50 @@
 import io.vertx.core.http.HttpMethod;
 import io.vertx.core.json.Json;
 import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
 import io.vertx.ext.web.handler.BodyHandler;
 import io.vertx.ext.web.handler.CorsHandler;
 import io.vertx.ext.web.handler.StaticHandler;
 
 public class AlohaVerticle extends AbstractVerticle {
+    private static final Brave BRAVE = new Brave.Builder("aloha")
+            .spanCollector(HttpSpanCollector.create(System.getenv("ZIPKIN_SERVER_URL"), new EmptySpanCollectorMetricsHandler()))
+            .build();
 
     @Override
     public void start() throws Exception {
         Router router = Router.router(vertx);
+        router.route().handler(ctx -> {
+            // note: this is *not* an example of how to properly integrate vert.x with zipkin
```

Comment:
Yes, there are several issues.

Zipkin (Brave) also makes the assumption 1 single thread is going to handle the request **and** this thread is dedicated to this request. Generally in Vert.x the first part is true, but here it's not because it use a blocking HTTP client so require a worker thread. The second part is NEVER true. 

As a consequence, Brave is not usable at all with Vert.x as it uses a thread locals. Something to remember: Never rely on thread local. It breaks a lots of case - especially when dealing with async (and not only Vert.x, async in general).

What @jpkrohling did is to use another container to propagate the parent span: the routing context. This approach would only works for HTTP handled with vert.x web, but not for anything else, where there are no such as container. That means explicit span propagation would be required.